calico_node/startup.py

```diff
@@ -266,7 +266,11 @@ def main():
         sys.exit(1)
 
     as_num = os.getenv("AS")
-    as_num = as_num or None
+    if os.getenv("USE_HOST_AS") == "true":
```

Comment:
> Currently, if calico is restarted with an empty ASN, client.create_host will explicitly delete the entry in etcd

yikes, that sounds like a bug to me! Shouldn't we be respecting any config that has been set when we start calico/node rather than blatting over it?

I'll leave up to @robbrockbank to make the final call, but I like being able to do this all off of one `AS` env var, and it feels like we should be respecting whatever config exists for the node when it is started.